drivers/gps/nrf9160_gps/nrf9160_gps.c

```diff
@@ -225,7 +243,28 @@ static void gps_thread(int dev_ptr)
 				goto wait;
 			}
 
-			LOG_ERR("recv() returned error: %d", len);
+			if (errno == EHOSTDOWN) {
+				LOG_DBG("GPS host is going down, sleeping");
+				k_delayed_work_cancel(&drv_data->timeout_work);
+				k_delayed_work_cancel(&drv_data->start_work);
+				atomic_clear(&drv_data->is_active);
+				atomic_set(&drv_data->is_shutdown, 1);
+				close(drv_data->socket);
```

Comment:
We use `nrf_socket` API here, so it should be `nrf_close()`. It makes difference.

Reply:
yes, that is true, there is no gnss socket in zephyr.